src/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -1484,10 +1517,13 @@ internal static unsafe void NumberToStringFormat(ref ValueStringBuilder sb, Numb
                                     else
                                     {
                                         sb.Append(ch); // Copy E or e to output
-                                        if (pFormat[src] == '+' || pFormat[src] == '-')
-                                            sb.Append(pFormat[src++]);
-                                        while (pFormat[src] == '0')
-                                            sb.Append(pFormat[src++]);
+                                        if (src < format.Length)
+                                        {
+                                            if (pFormat[src] == '+' || pFormat[src] == '-')
+                                                sb.Append(pFormat[src++]);
+                                            while (src < format.Length && pFormat[src] == '0')
+                                                sb.Append(pFormat[src++]);
```

Comment:
It'd be great if you could add all the braces which should be in this method per the style guidelines, especially with all the `&&`s that are being added.

Reply:
Ok